lib/Doctrine/DBAL/Types/StringType.php

```diff
@@ -19,14 +19,6 @@ public function getSQLDeclaration(array $fieldDeclaration, AbstractPlatform $pla
         return $platform->getVarcharTypeDeclarationSQL($fieldDeclaration);
     }
 
-    /**
-     * {@inheritdoc}
-     */
-    public function getDefaultLength(AbstractPlatform $platform)
-    {
-        return $platform->getVarcharDefaultLength();
```

Comment:
Speaking of which, why do we need to know the default type length on the platform level? E.g.:

https://github.com/doctrine/dbal/blob/caf55ee7df850234e21c257fdc63d9be00e02a98/lib/Doctrine/DBAL/Platforms/AbstractPlatform.php#L298-L300

We're taking our hard-coded value and passing it explicitly to the field declaraion. Would it make sense to leave the length unspecified if the developer doesn't care? Otherwise, it leads to the pointless expctations like https://github.com/doctrine/dbal/pull/3133/commits/e2e0de250b078119dd19c3926bb8eaedc9501a4d#diff-21fd51266275832f478e50a5b1f22748L219.

Reply:
Completely unspecified length is probably better: should be explicitly specified by developers anyway.

Reply:
How portalbe is that? From what I remember MySQL requires maximum length when declaring VARCHAR columns while PostgreSQL doesn't.

Reply:
There are two points here:

1. If a platform doesn't require max length (PostgreSQL), DBAL shouldn't require max length either but it shouldn't provide any hard-coded length in DDL. It should omit it.
2. If a platform does require MySQL, DBAL should require it in the column definition but it shouldn't provide any hard-coded length in DDL. It should fail.

This way, if you use only one platform (PostgreSQL) and don't care about portability, you can omit the length and use DBAL w/o any artificial restrictions. Otherwise, you have to explicitly specify the length. It's not the DBAL's job to specify it for you.